packages/expo-cli/src/credentials/views/AndroidKeystore.ts

```diff
@@ -0,0 +1,234 @@
+    log.warn(
+      chalk.bold(
+        'Android keystores must be identical to the one previously used to submit your app to the Google Play Store.'
+      )
+    );
+    log.warn(
+      'Please read https://docs.expo.io/distribution/building-standalone-apps/#if-you-choose-to-build-for-android for more info before proceeding.'
+    );
+    log.newLine();
+    log.warn(
+      chalk.bold('Your keystore will be backed up to your current directory if you continue.')
+    );
+    log.newLine();
+  }
+}
+
+interface DownloadKeystoreOptions {
+  quiet?: boolean;
+  displayCredentials?: boolean;
+  outputPath?: string;
+}
+
+class DownloadKeystore implements IView {
+  constructor(private experienceName: string, private options?: DownloadKeystoreOptions) {}
+
+  async open(ctx: Context): Promise<IView | null> {
+    const { confirm } = await prompt({
+      type: 'confirm',
+      name: 'confirm',
+      message: 'Do you want to display the Android Keystore credentials?',
+      when: () => this.options?.displayCredentials === undefined && !this.options?.quiet,
+    });
+
+    const keystoreObj = await ctx.android.fetchKeystore(this.experienceName);
+
+    const { keystore, keystorePassword, keyAlias, keyPassword }: any = keystoreObj || {};
+    if (!keystore || !keystorePassword || !keyAlias || !keyPassword) {
+      if (!this.options?.quiet) {
+        log.warn('There is no valid Keystore defined for this app');
+      }
+      return null;
+    }
+
+    const keystorePath =
+      this.options?.outputPath ?? `${this.experienceName.replace('/', '__')}.bak.jks`;
+
+    await maybeRenameExistingFile(ctx.projectDir, keystorePath);
+    if (!this.options?.quiet) {
+      log(chalk.green(`Saving Keystore to ${keystorePath}`));
+    }
+    const storeBuf = Buffer.from(keystore, 'base64');
+    await fs.writeFile(keystorePath, storeBuf);
+
+    if (this.options?.displayCredentials ?? confirm) {
+      log(`Keystore credentials
+  Keystore password: ${chalk.bold(keystorePassword)}
+  Key alias:         ${chalk.bold(keyAlias)}
+  Key password:      ${chalk.bold(keyPassword)}
+
+  Path to Keystore:  ${keystorePath}
+      `);
+    }
+    return null;
+  }
+}
+
+async function getKeystoreFromParams(options: {
+  keystorePath?: string;
+  keystoreAlias?: string;
+}): Promise<Keystore | null> {
+  const keyAlias = options.keystoreAlias;
+  const keystorePath = options.keystorePath;
+  const keystorePassword = process.env.EXPO_ANDROID_KEYSTORE_PASSWORD;
+  const keyPassword = process.env.EXPO_ANDROID_KEY_PASSWORD;
+  if (!keyAlias && !keystorePath) {
+    return null;
+  }
+
+  if (!keystorePath || !keyAlias || !keystorePassword || !keyPassword) {
+    console.log(keystorePath, keyAlias, keystorePassword, keyPassword);
+    throw new Error(
+      'In order to provide a Keystore through the CLI parameters, you have to pass --keystore-alias, --keystore-path parameters and set EXPO_ANDROID_KEY_PASSWORD and EXPO_ANDROID_KEYSTORE_PASSWORD environment variables.'
```

Comment:
Hmmm, we need to be consistent about the capitalization of `Keystore`. In some places, it's not capitalized and in some places it is.